API.adoc

```diff
@@ -0,0 +1,402 @@
+added as Metadata for this Thing.
+
+*Example 1*
+[source,json]
+----
+{
+    "urn": "urn:thing:uuid:346e742e-2f1e-4d91-9ffe-7b38eec6219c",
+    "active": true,
+    "name": "My home",
+    "description": "Where I live right now",
+    "large": true
+}
+----
+
+.Response
+----
+201 CREATED
+----
+[source,json]
+----
+{
+    "urn": "urn:thing:uuid:346e742e-2f1e-4d91-9ffe-7b38eec6219c",
+    "type": "Building",
+    "tenantUrn": "urn:tenant:uuid:69bb7c6a-a43b-493d-8e9d-e5a3ed65728a",
+    "active": true
+}
+----
+
+*Example 2*
+[source,json]
+----
+{
+}
+----
+
+.Response
+----
+201 CREATED
+----
+[source,json]
+----
+{
+    "urn": "urn:thing:uuid:2519a8ba-fadf-4a85-a965-5a59a5b43e7d",
+    "type": "Building",
+    "tenantUrn": "urn:tenant:uuid:69bb7c6a-a43b-493d-8e9d-e5a3ed65728a",
+    "active": true
+}
+----
+
+[[update1]]
+==== Update - `PUT /{type}/{urn}`
+
+Update an already existing Thing belonging to the Tenant of the authenticated User,
+and update its associated Metadata.
+
+NOTE: Type, URN, and Tenant URN are immutable for an existing Thing; this endpoint
+is only useful for turning the _active_ flag off and on for the Thing itself, and
+for updating the Thing's Metadata.
+
+----
+PUT /Building/urn:thing:uuid:346e742e-2f1e-4d91-9ffe-7b38eec6219c
+----
+[source,json]
+----
+{
+    "active": false,
+    "name": "My previous home"
+}
+----
+.Response
+----
+204 NO CONTENT
+----
+
+
+[[read1]]
+==== Read - `GET /{type}?fields={keyNames}&page={pageNumber}&size={pageSize}`
+
+Get all Things of the specified type belonging to the Tenant of the authenticated
+User, and their associated Metadata. If the `fields` query parameter is set,
+the result will include only the Metadata fields specified. If the `fields`
+query parameter is not set, all Metadata for the returned Things will be included
+in the result. Results are paged.
+
+----
+GET /Building?fields=name,description,large&page=0&size=100
+----
+.Response
+----
+200 OK
+----
+[source,json]
+----
+{
+    "data": [
+        {
+            "urn": "urn:thing:uuid:346e742e-2f1e-4d91-9ffe-7b38eec6219c",
+            "type": "Building",
+            "tenantUrn": "urn:tenant:uuid:69bb7c6a-a43b-493d-8e9d-e5a3ed65728a",
+            "active": true,
+            "name": "My home",
+            "description": "My home in US",
+            "large": true
+        },
+        {
+            "urn": "urn:thing:uuid:2519a8ba-fadf-4a85-a965-5a59a5b43e7d",
+            "type": "Building",
+            "tenantUrn": "urn:tenant:uuid:69bb7c6a-a43b-493d-8e9d-e5a3ed65728a",
+            "active": true,
+            "name": "My school",
+            "description": "My school in US"
+        }
+    ],
+    "page" : {
+        "size" : 100,
+        "totalElements" : 2,
+        "totalPages" : 1,
+        "number" : 0
+    }
+}
+----
+
+
+[[read2]]
+==== Read - `GET /{type}/{urn}?fields={keyNames}`
+
+Get the Thing with the the specified type and URN belonging to the Tenant of the
```

Comment:
This one I disagree on, since it introduces ambiguity by separating type/URN from Thing.